internal/service/alarms/internal/alertmanager/alertmanager.go

```diff
@@ -63,3 +68,114 @@ func Setup(ctx context.Context, cl client.Client) error {
 	slog.Info("Successfully configured alertmanager")
 	return nil
 }
+
+// ConvertAmToAlarmEventRecordModels get alarmEventRecords based on the alertmanager notification and AlarmDefinition
+func ConvertAmToAlarmEventRecordModels(am *api.AlertmanagerNotification, aDefinitionRecords []models.AlarmDefinition, clusterIDToObjectTypeID map[uuid.UUID]uuid.UUID) []models.AlarmEventRecord {
+	records := make([]models.AlarmEventRecord, 0, len(am.Alerts))
+	for _, alert := range am.Alerts {
+		slog.Info("Converting Alertmanager alert", "alert name", GetAlertName(*alert.Labels))
+		oranPS, _ := GetPerceivedSeverity(*alert.Labels)
+		record := models.AlarmEventRecord{
+			AlarmRaisedTime:   *alert.StartsAt,
+			AlarmClearedTime:  alert.EndsAt,
+			AlarmChangedTime:  alert.StartsAt, //Note: this value is not available as part of the notification (only in v2 api as updatedAt)
+			PerceivedSeverity: finalOranSeverity(oranPS, *alert.Status),
+			AlarmStatus:       string(*alert.Status),
+			Fingerprint:       *alert.Fingerprint,
+			ObjectID:          GetObjectID(*alert.Labels),
+			ObjectTypeID:      clusterIDToObjectTypeID[GetClusterID(*alert.Labels)],
+		}
+
+		// Update Extensions with things we didn't really process
+		record.Extensions = getExtensions(*alert.Labels, *alert.Annotations)
+
+		// See if possible to pick up additional info from its definition
+		for _, def := range aDefinitionRecords {
+			if def.AlarmName == GetAlertName(*alert.Labels) && def.ObjectTypeID == record.ObjectTypeID && severityToPerceivedSeverity(def.Severity) == record.PerceivedSeverity {
+				record.AlarmDefinitionID = def.AlarmDefinitionID
+				record.ProbableCauseID = def.ProbableCauseID
+			}
+		}
+
+		// Anything else that's not mentioned explicitly will be handled by DB such ID generation and default values as needed.
+		records = append(records, record)
+	}
+
+	return records
+}
+
+// finalOranSeverity set severity cleared
+func finalOranSeverity(ps api.PerceivedSeverity, status api.AlertmanagerNotificationStatus) api.PerceivedSeverity {
```

Comment:
I don't think you need another level of abstraction for this. Simply after record

Reply:
removed